labgrid/driver/consoleexpectmixin.py

```diff
@@ -54,6 +55,17 @@ def expect(self, pattern, timeout=-1):
         index = self._expect.expect(pattern, timeout=timeout)
         return index, self._expect.before, self._expect.match, self._expect.after
 
+    @Driver.check_active
+    @step(args=['quiet_time'])
+    def settle(self, quiet_time, timeout=None):
+        start = time.time()
+        try:
+            while self.read(timeout=quiet_time):
+                if timeout is not None and time.time() > start + timeout:
+                    raise pexpect.TIMEOUT("Timeout of {} seconds exceeded waiting for output to settle".format(timeout))
```

Comment:
Use the `Timeout` helper class from labgrid.util, which also uses `time.monotonic` instead of `time.time()` (which is can have discontinuities).